libs/ui-components/src/components/common/DisplayName.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { CopyIcon } from '@patternfly/react-icons/dist/js/icons/copy-icon';
+import { Button, Icon, Tooltip } from '@patternfly/react-core';
+
+import { useTranslation } from '../../hooks/useTranslation';
+import { Link, type RouteWithPostfix } from '../../hooks/useNavigate';
+
+import './DisplayName.css';
+
+const maxDisplayLength = 50;
+
+type DisplayNameProps = {
+  name: string;
```

Comment:
could be 
```suggestion
  name: string | undefined;
```

since `getDisplayName` also accepts undefined..it will also cleanup the component's prop as we wont have to do `as string` or `value || '-'` etc.

Reply:
Good idea, changed it.